extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +184,123 @@ private void restartJobs() {
         if (schedulerFuture != null && !schedulerFuture.isCancelled()) {
             schedulerFuture.cancel(true);
         }
-
-        schedulerFuture = scheduler.schedule(new Runnable() {
-            @Override
-            public void run() {
+        schedulerFuture = scheduler.schedule(() -> {
+            monitor.lock();
+            try {
                 stopJobs();
-
-                try {
-                    synchronized (schedulerLock) {
-                        if (quartzScheduler == null) {
-                            quartzScheduler = StdSchedulerFactory.getDefaultScheduler();
+                if (getThing().getStatus() == ONLINE) {
+                    String thingUID = getThing().getUID().toString();
+                    String typeId = getThing().getThingTypeUID().getId();
+                    // Daily Job
+                    Job dailyJob = getDailyJob();
+                    if (dailyJob == null) {
+                        logger.error("Daily Job instance not present");
+                        return;
+                    }
+                    if (scheduledExecutor == null) {
+                        logger.warn("Thread Pool Executor is not available");
+                        return;
+                    }
+                    Expression midNightExpression = new CronExpression(DAILY_MIDNIGHT);
+                    if (addJobToQueue(dailyJob)) {
+                        scheduledExecutor.schedule(dailyJob, midNightExpression);
+                    }
+                    logger.info("Scheduled astro job-daily-{} at midnight for thing {}", typeId, thingUID);
+
+                    // Startup Job
+                    LocalDateTime currentTime = LocalDateTime.now();
+                    Expression cronExpression = new CronExpression(createCronFromTemporal(currentTime));
```

Comment:
This does not work, this job will never be executed, because by the time the framework is ready to process the expression, the scheduled time lies in the past. Please call `dailyJob.run()` directly here to get the information immediately instead of waiting until midnight.